Cython/Distutils/build_ext.py

```diff
@@ -10,34 +11,126 @@
 else:
     from distutils.command.build_ext import build_ext as _build_ext
 
+try:
+    from __builtin__ import basestring
+except ImportError:
+    basestring = str
 
-class new_build_ext(_build_ext, object):
 
+class build_ext(_build_ext, object):
+
+    sep_by = _build_ext.sep_by
     user_options = _build_ext.user_options[:]
     boolean_options = _build_ext.boolean_options[:]
 
     user_options.extend([
+        ('cython-cplus', None,
+             "generate C++ source files"),
+        ('cython-create-listing', None,
+             "write errors to a listing file"),
+        ('cython-line-directives', None,
+             "emit source line directives"),
+        ('cython-include-dirs=', None,
+             "path to the Cython include files" + sep_by),
         ('cython-c-in-temp', None,
              "put generated C files in temp directory"),
+        ('cython-gen-pxi', None,
+            "generate .pxi file for public declarations"),
+        ('cython-directives=', None,
+            "compiler directive overrides"),
+        ('cython-gdb', None,
+             "generate debug information for cygdb"),
+        ('cython-compile-time-env', None,
+            "cython compile time environment"),
         ])
 
     boolean_options.extend([
-        'cython-c-in-temp'
+        'cython-cplus', 'cython-create-listing', 'cython-line-directives',
+        'cython-c-in-temp', 'cython-gdb',
     ])
 
     def initialize_options(self):
         _build_ext.initialize_options(self)
+        self.cython_cplus = 0
+        self.cython_create_listing = 0
+        self.cython_line_directives = 0
+        self.cython_include_dirs = None
+        self.cython_directives = None
         self.cython_c_in_temp = 0
+        self.cython_gen_pxi = 0
+        self.cython_gdb = False
+        self.cython_compile_time_env = None
+
+    def finalize_options(self):
+        _build_ext.finalize_options(self)
+        if self.cython_include_dirs is None:
+            self.cython_include_dirs = []
+        elif isinstance(self.cython_include_dirs, basestring):
+            self.cython_include_dirs = \
+                self.cython_include_dirs.split(os.pathsep)
+        if self.cython_directives is None:
+            self.cython_directives = {}
 
     def build_extension(self, ext):
         from Cython.Build.Dependencies import cythonize
-        if self.cython_c_in_temp:
+
+        # Set up the include_path for the Cython compiler:
+        #    1.    Start with the command line option.
+        #    2.    Add in any (unique) paths from the extension
+        #        cython_include_dirs (if Cython.Distutils.extension is used).
+        #    3.    Add in any (unique) paths from the extension include_dirs
+        includes = list(self.cython_include_dirs)
+        try:
+            for i in ext.cython_include_dirs:
+                if i not in includes:
+                    includes.append(i)
```

Comment:
The variable name `i` is commonly reserved for integer loop variables, suggesting a misleading context here.
```suggestion
            for include_dir in ext.cython_include_dirs:
                if include_dir not in includes:
                    includes.append(include_dir)
```